enhancements/sig-architecture/225-clusteradm-operator/README.md

```diff
@@ -0,0 +1,557 @@
+# Clusteradm Operator
+
+## Release Signoff Checklist
+
+- [] Enhancement is `provisional`
+- [] Design details are appropriately documented from clear requirements
+- [] Test plan is defined
+- [] Graduation criteria for dev preview, tech preview, GA
+- [] User-facing documentation is created in [website](https://github.com/open-cluster-management-io/open-cluster-management-io.github.io/)
+
+## Summary
+
+This proposal introduces the clusteradm operator, which will provide a declarative interface for managing multi-cluster environments. It wraps the imperative `clusteradm` CLI in a Kubernetes-native way, introducing the `MultiCluster` CRD to enable automated bootstrapping of Kubernetes clusters into a multicluster topology. Ongoing maintenance of the multi-cluster in the form of configuration patches and OCM version upgrades is also supported.
+
+## Motivation
+
+OCM's current installation procedure is imperative-only, requiring multiple manual operations, either via `clusteradm` or direct application of OCM custom resources to various Kubernetes clusters. The lack of a declarative interface for managing the lifecycle of an OCM multi-cluster has several limitations:
+
+- Complexity around bootstrapping and maintenance
+- Multi-cluster topology / configuration cannot be managed via GitOps
+- Not aligned with Kubernetes best practices
+
+### Goals
+
+- Declarative lifecycle management of an OCM multi-cluster
+- Simplify and streamline the OCM onboarding and maintenance experience
```

Comment:
Users will need to install this clusteradm operator and API as the first step. Would you mind adding a bit more detail about how that installation would work in the KEP? For example, will it be via a Helm chart, a new clusteradm subcommand, or something else?

Reply:
The operator we've developed is currently installed via Helm. We could certainly extend `clusteradm` to support installing that Helm chart as well. Maybe, `clusteradm fleet-operator-init`? Any command name proposals?

Reply:
Helm chart sounds good. We probably don't need the clusteradm sub-command for a while.